test/e2e/dnslookup_test.go

```diff
@@ -0,0 +1,299 @@
+		out, err := cmdRunWithOutput("wsl", "--distribution", wslDistroName, "--exec", "/bin/wslpath", ".")
+		if err != nil {
+			return false
+		}
+		// remove all the weirdness from WSL output
+		return strings.TrimSpace(out) == "."
+	}, tryInterval, timeout)
+	require.NoErrorf(t, err, "Failed to check if %v wsl distro is running", wslDistroName)
+
+	dnsInfs, err := helper.GetDNSInterfaces()
+	require.NoError(t, err, "Failed getting DNS addrs associated to interfaces")
+
+	// This is to cache all the exsisting DNS addresses
+	guidToDNSAddr, err := cacheExsitingDNSAddrs(dnsInfs)
+	require.NoError(t, err, "Failed caching exsisting DNS server addresses")
+
+	// restore the system DNS servers to the original state
+	defer restoreSystemDNS(t, dnsInfs, guidToDNSAddr)
+
+	t.Log("Updating network interfaces with test DNS server addr")
+	// Update the dns addrs to test server
+	updateSystemDNS(t, dnsInfs)
+
+	t.Log("Building host-resolver host binary")
+	err = buildBinaries("../../...", "windows", tmpDir)
+	require.NoError(t, err, "Failed building host-resolver.exe")
+
+	t.Log("Building host-resolver peer binary")
+	err = buildBinaries("../../...", "linux", tmpDir)
+	require.NoError(t, err, "Failed building host-resolver")
+
+	aRecords := testdns.LoadRecords("../testdata/test-300.csv")
+
+	tcpHandler := testdns.NewHandler(false)
+	tcpHandler.Arecords = aRecords
+
+	udpHandler := testdns.NewHandler(true)
+	udpHandler.Arecords = aRecords
+
+	testServer := testdns.Server{
+		Addr:       testSrvAddr,
+		TCPPort:    dnsPort,
+		UDPPort:    dnsPort,
+		TCPHandler: tcpHandler,
+		UDPHandler: udpHandler,
+	}
+	t.Log("Starting test upstream DNS server")
+	go testServer.Run()
+
+	t.Logf("Starting host-resolver peer process in wsl [%v]", wslDistroName)
+	peerCmd := cmdExec(
+		tmpDir,
+		"wsl", "--user", "root",
+		"--distribution", wslDistroName,
+		"--exec", "./rancher-desktop-host-resolver", "vsock-peer")
+	err = peerCmd.Start()
+	require.NoError(t, err, "Starting host-resolver peer process faild")
+	defer func() {
+		_ = peerCmd.Process.Kill()
+	}()
+
+	t.Log("Starting host-resolver host process")
+	resolverExecPath := fmt.Sprintf("%v/rancher-desktop-host-resolver.exe", tmpDir)
+	hostCmd := cmdExec(
+		tmpDir,
+		resolverExecPath, "vsock-host",
+		"--upstream-servers", fmt.Sprintf("[%v]", testSrvAddr))
+	err = hostCmd.Start()
+	require.NoError(t, err, "Starting host-resolver host process faild")
+	defer func() {
+		_ = hostCmd.Process.Kill()
+	}()
+
+	t.Log("Building DNS hammer binary")
+	err = buildBinaries("../...", "linux", tmpDir)
+	require.NoError(t, err, "Failed building dnsHammer")
+
+	err = copyTestData("../testdata/test-300.csv", fmt.Sprintf("%s/test-300.csv", tmpDir))
+	require.NoError(t, err, "Failed copying test data file")
+
+	t.Log("Confirming host-resolver peer process is up")
+	peerCmdTimeout := time.Second * 10
+	confirm(func() bool { //nolint:errcheck // we don't care about the error
+		p, _ := os.FindProcess(peerCmd.Process.Pid)
```

Comment:
If there was an error (maybe the pid doesn't exist), then `p` would be `nil`, so accessing `p.Pid` on the next line would panic?